job-service-db/src/main/resources/procedures/getJobs.sql

```diff
@@ -183,6 +184,46 @@ BEGIN
          ELSE quote_ident(in_sort_field)
        END ||
         ' ' || CASE WHEN in_sort_ascending THEN 'ASC' ELSE 'DESC' END;
-    RETURN QUERY EXECUTE sql;
+
+    -- Create temporary table as a base to update the job progress
+    EXECUTE 'CREATE TEMPORARY TABLE get_job_temp ON COMMIT DROP AS ' || sql;
+
+    ALTER TABLE get_job_temp ADD COLUMN id SERIAL PRIMARY KEY;
+
+    -- Create a duplicate of that temporary table to store the results after the update
+    CREATE TEMPORARY TABLE new_table ON COMMIT DROP AS SELECT * FROM get_job_temp ORDER BY id;
+
+    FOR jobId IN SELECT * FROM get_job_temp LOOP
+        -- Take out an exclusive update lock on the job row
```

Comment:
Minor issue but all these statements will be run multiple times for each job that has > 1 label.